src/tir/transforms/lower_sparse_tir.cc

```diff
@@ -421,18 +422,22 @@ class IndexTransformer : public StmtExprMutator {
           auto sf_axis = axis.as<SparseFixedAxisNode>();
           PrimExpr l, r;
           std::tie(l, r) = sp_buf_ctx_.GetIndicesRange(dim);
-          offset = lower_bound(sf_axis->indices->data, coordinate, l, r);
+          offset = lower_bound(sf_axis->indices->data, coordinate, l, r) - l;
           break;
         }
         case AxisKind::kSparseVariable:
           auto sv_axis = axis.as<SparseVariableAxisNode>();
           PrimExpr l, r;
           std::tie(l, r) = sp_buf_ctx_.GetIndicesRange(dim);
-          offset = lower_bound(sv_axis->indices->data, coordinate, l, r);
+          offset = lower_bound(sv_axis->indices->data, coordinate, l, r) - l;
           break;
       }
     }
 
+    // update offset
+    PrimExpr new_offset = AggregateOffset(sp_buf_ctx_.top()->offsets_.back(), axis,
```

Comment:
Why don't we expose a function to manipulate the offset?
I suggest not operating on `top()` directly.